python/cudf/cudf/io/csv.py

```diff
@@ -158,33 +162,48 @@ def read_csv(
             header = 0
 
     hex_cols: list[abc.Hashable] = []
-    new_dtypes: list[plc.DataType] | dict[abc.Hashable, plc.DataType] = []
+    cudf_dtypes: list[DtypeObj] | dict[abc.Hashable, DtypeObj] | DtypeObj = []
+    plc_dtypes: list[plc.DataType] | dict[abc.Hashable, plc.DataType] = []
     if dtype is not None:
         if isinstance(dtype, abc.Mapping):
-            new_dtypes = {}
+            plc_dtypes = {}
+            cudf_dtypes = {}
             for k, col_type in dtype.items():
-                if is_hashable(col_type) and col_type in _CSV_HEX_TYPE_MAP:
+                if isinstance(col_type, str) and col_type in _CSV_HEX_TYPE_MAP:
                     col_type = _CSV_HEX_TYPE_MAP[col_type]
                     hex_cols.append(str(k))
 
-                new_dtypes[k] = _get_plc_data_type_from_dtype(
-                    cudf.dtype(col_type)
-                )
-        elif cudf.api.types.is_scalar(dtype) or isinstance(
-            dtype, (np.dtype, pd.api.extensions.ExtensionDtype, type)
+                cudf_dtype = cudf.dtype(col_type)
+                cudf_dtypes[k] = cudf_dtype
+                plc_dtypes[k] = _get_plc_data_type_from_dtype(cudf_dtype)
+        elif isinstance(dtype, str) or isinstance(
+            dtype,
+            (
+                np.dtype,
+                pd.api.extensions.ExtensionDtype,
+                cudf.core.dtypes._BaseDtype,
+                type,
+            ),
         ):
```

Comment:
Can we collapse this into a single isinstance call instead of two?

Reply:
Thanks, good catch.